### PR TITLE
handle any zero-sized Data in http:data with fin

### PR DESCRIPTION
**Pull request by mbj4668**
I passed `[]` as Data to gun:data/4 with `fin`, and got really weird results...
Eventually I found this.

### CODE REVIEW

**Comment by essen** [comment]
Please add a test if possible.

File: test/gun_SUITE.erl [hunk 1]
```diff
@@ -107,6 +107,21 @@ ignore_empty_data_http(_) ->
...
+	{ok, Pid} = gun:open("localhost", OriginPort),
+	{ok, http} = gun:await_up(Pid),
+	handshake_completed = receive_from(OriginPid),
+	Ref = gun:post(Pid, "/", []),
+	gun:data(Pid, Ref, nofin, "hello"),
+	gun:data(Pid, Ref, fin, ""),
+	Data = receive_all_from(OriginPid, 500),
+	Lines = binary:split(Data, <<"\r\n">>, [global]),
+	Zero = [Z || <<"0">> = Z <- Lines],
+	1 = length(Zero),
```
Comment (essen) [inline test/gun_SUITE.erl:122]: Took me a while to figure out you were looking for 0\r\n. Maybe search for `<<"\r\n0\r\n">>` via `binary:match` instead.

Comment (mbj4668) [reply test/gun_SUITE.erl:122]: I copied the existing test case `ignore_empty_data_http`.
W/o the fix, the code sent two "0".  This test ensures it just gets one.

Comment (essen) [reply test/gun_SUITE.erl:122]: Even better if you can fix that test as well.

Comment (essen) [reply test/gun_SUITE.erl:122]: Perhaps instead of doing this it would be better to split in two at `\r\n\r\n` and try to parse the body.

**Review by Payaspx — APPROVED** [review]

**Pull request closed without merging**